### ISSUE TITLE
Syntax error in an importing root file suppresses diagnostics in unrelated root files

### ISSUE THREAD

**Issue by mohsen1** [microsoft/TypeScript#63735]
### 🔎 Search Terms

`parse error`, `relative import`, `missing diagnostics`, `root files`, `TS1003`, `TS2322`

### 🕗 Version & Regression Information

This occurs in every version tested: 7.0.2 and `7.1.0-dev.20260809.1` (nightly). I searched open and closed issues.

### 💻 Code

```ts
// target-clean.ts
export const foo = 123;

// importer-with-syntax-error.ts
import { foo as "bad" } from "./target-clean";

// unrelated-error.ts
const q: string = 12345;
```

```sh
tsc --noEmit --strict target-clean.ts importer-with-syntax-error.ts unrelated-error.ts
```

### 🙁 Actual behavior

Only TS1003 is reported. The unrelated file's TS2322 is silently omitted. Removing the relative import or replacing the syntax error with a semantic error restores TS2322.

### 🙂 Expected behavior

Both TS1003 and TS2322 are reported.

**Comment by Mahnoor-Zaffar** [comment]
Hey! Thanks for reporting this.


The most likely root cause is related to how the TypeScript compiler processes files with errors. It may halt further diagnostics when encountering a syntax error in a root file, thereby skipping checks in other files.

To address this issue, it would be reasonable to investigate the compiler's error handling logic, particularly how it manages the compilation process when errors are present in multiple files. It would be helpful to know if there are specific configurations or flags being used during compilation that could affect error reporting.

If no one is currently working on this, I'd be happy to take a look and put together a fix.

**Comment by RyanCavanaugh** [comment]
This is intentional behavior. A syntax error can radically alter the interpretation of everything else in a file, so showing you potentially hundreds of type errors when the only thing you need to do is delete a stray `}` would be very bad user experience.
